crates/dekaf/src/log_journal.rs

```diff
@@ -0,0 +1,344 @@
+    ) -> anyhow::Result<()> {
+        let mut log_data = Vec::new();
+
+        let (ops_logs_journal_client, ops_logs_journal) = loop {
+            match logs_rx.recv().await {
+                Some(LoggingMessage::SetTaskName(name)) => {
+                    let (client, ops_logs) = self.get_journal_client(name).await?;
+                    break (client, ops_logs);
+                }
+                Some(LoggingMessage::Log(log)) => {
+                    log_data.append(&mut self.serialize_log(log));
+                }
+                Some(LoggingMessage::Shutdown) | None => return Ok(()),
+            }
+        };
+
+        self.append_log_data(
+            log_data.into(),
+            ops_logs_journal.as_str(),
+            &ops_logs_journal_client,
+        )
+        .await?;
+
+        while let Some(msg) = logs_rx.recv().await {
+            match msg {
+                LoggingMessage::SetTaskName(_) => {}
+                LoggingMessage::Log(log) => {
+                    self.append_log_data(
+                        self.serialize_log(log).into(),
+                        ops_logs_journal.as_str(),
+                        &ops_logs_journal_client,
+                    )
+                    .await?;
+                }
+                LoggingMessage::Shutdown => break,
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn append_log_data(
+        &self,
+        log_data: Bytes,
+        journal: &str,
+        client: &journal::Client,
+    ) -> anyhow::Result<()> {
+        let resp = client.append(
+            gazette::broker::AppendRequest {
+                journal: journal.to_owned(),
+                ..Default::default()
+            },
+            || {
+                futures::stream::once({
+                    let value = log_data.clone();
+                    async move { Ok(value) }
+                })
+            },
+        );
+
+        tokio::pin!(resp);
+
+        let mut attempts = 0;
+
+        loop {
+            match resp.try_next().await {
+                Err(err) if err.is_transient() && attempts < 3 => {
+                    attempts += 1;
+                }
+                Err(err) => {
+                    if err.is_transient() {
+                        tracing::warn!(
+                            ?err,
+                            attempts,
+                            "Got recoverable error multiple times while trying to write logs"
+                        )
+                    }
+                    return Err(err.into());
+                }
+                Ok(_) => return Ok(()),
+            }
+        }
+    }
+
+    fn serialize_log(&self, log: ops::Log) -> Vec<u8> {
+        let uuid = gazette::uuid::build(
+            self.producer,
+            gazette::uuid::Clock::from_time(std::time::SystemTime::now()),
+            uuid::Flags(
+                (message_flags::MASK | message_flags::OUTSIDE_TXN)
```

Comment:
@jgraettinger do these looks right to you? IIRC I was seeing some suspicious `Unexpected UUID flags` errors in my Gazette logs, but I'm not able to find/reproduce that